### PR TITLE
Simplify handling connectReply stream

### PR DESCRIPTION
**Pull request by akrambek**

### CODE REVIEW

File: src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientConnectReplyStream.java [hunk 1]
```diff
@@ -328,15 +310,22 @@ private void handleDataWhenNotBuffering(
...
             final OctetsFW payload = data.payload();
-            final int limit = payload.limit();
-            int offset = payload.offset();
-
-            offset = decode(payload.buffer(), offset, limit);
+            if (slotIndex == NO_SLOT)
+            {
+                final int limit = payload.limit();
+                int offset = payload.offset();
+                offset = decode(payload.buffer(), offset, limit);
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientConnectReplyStream.java]: Suggest the following:
```
                DirectBuffer buffer = payload.buffer();
                int offset = payload.offset();
                int limit = payload.limit();

                if (slotIndex != NO_SLOT)
                {
                    final MutableDirectBuffer slotBuffer = bufferPool.buffer(slotIndex);
                    slotBuffer.putBytes(slotOffset, buffer, offset, limit - offset);
                    slotOffset += limit - offset;
                    buffer = slotBuffer;
                    offset = 0;
                    limit = slotOffset;
                }

                DecoderState previous = null;
                while (offset <= limit && previous != decodeState)
                {
                    previous = decodeState;
                    offset = decodeState.decode(buffer, offset, limit);
                }

                if (offset < limit)
                {
                    if (slotIndex == NO_SLOT)
                    {
                        slotIndex = factory.bufferPool.acquire(connectReplyId);
                    }

                    if (slotIndex == NO_SLOT)
                    {
                        factory.writer.doReset(connectReplyThrottle, connectRouteId, connectReplyId, factory.supplyTrace.getAsLong());
                        connection.persistent = false;
                        doCleanup(CloseAction.ABORT);
                    }
                    else
                    {
                        final MutableDirectBuffer slotBuffer = bufferPool.buffer(slotIndex);
                        slotBuffer.putBytes(0, buffer, offset, limit - offset);
                        slotOffset = limit - offset;
                    }
                }
                else if (slotIndex != NO_SLOT)
                {
                    factory.bufferPool.release(slotIndex);
                    slotIndex = NO_SLOT;
                    slotOffset = 0;
                }
```
... and remove `slotPosition` as it is no longer needed.

This should be the only method to deal with `slotIndex`, except for cleaning up on `ABORT` or `RESET`.

File: src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientConnectReplyStream.java [hunk 2]
```diff
@@ -130,7 +130,7 @@ private void handleStreamBeforeBegin(
         }
     }
 
-    private void handleStreamWhenBuffering(
+    private void handleStream(
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientConnectReplyStream.java]: Add `BeginFW.TYPE_ID` to `switch`, remove `streamState` and receive all stream frames directly in `handleStream`.

File: src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientConnectReplyStream.java [hunk 3]
```diff
@@ -102,96 +100,42 @@ public String toString()
...
             final EndFW end = this.factory.endRO.wrap(buffer, index, index + length);
-            handleEnd(end);
+            if (slotIndex == NO_SLOT)
+            {
+                handleEnd(end);
+            }
+            else
+            {
+                handleEndWhenBuffering(buffer, index, length);
+            }
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientConnectReplyStream.java]: Please simplify this to call `handleEnd(end)` unconditionally, then suggest modifying `handleEnd` as follows:
```
    private void handleEnd(
        EndFW end)
    {
        final long streamId = end.streamId();
        assert streamId == connectReplyId;

        if (slotIndex != NO_SLOT && (responseState == BEFORE_HEADERS || responseState == DATA))
        {
            endDeferred = true;
        }
        else
        {
            // existing handleEnd implementation
        }
    }
```
...and remove `handleEndWhenBuffering` as it is no longer needed.

File: src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientConnectReplyStream.java [hunk 4]
```diff
@@ -958,12 +807,21 @@ private void handleWindow(
 
         if (slotIndex != NO_SLOT)
         {
-            decodeBufferedData();
+            MutableDirectBuffer slot = factory.bufferPool.buffer(slotIndex);
+            decode(slot, 0, slotOffset);
+            if (slotOffset == 0 && endDeferred)
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientConnectReplyStream.java]: This should really be `if (slotIndex == NO_SLOT && endDeferred)`, right?

Comment (akrambek) [reply src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientConnectReplyStream.java]: I guess that works too.

File: src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientConnectReplyStream.java [hunk 5]
```diff
@@ -102,70 +100,31 @@ public String toString()
...
-        this.streamState = this::handleStreamBeforeBegin;
         this.throttleState = this::handleThrottleBeforeBegin;
         this.windowHandler = this::handleWindow;
     }
 
     @Override
     public void accept(int msgTypeId, DirectBuffer buffer, int index, int length)
     {
-        streamState.accept(msgTypeId, buffer, index, length);
+        handleStream(msgTypeId, buffer, index, length);
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientConnectReplyStream.java]: Use `::handleStream` directly and remove `implements MessageConsumer`.

**Pull request merged**